### ISSUE TITLE
OOM on 3090 while running the training of NeRF

### ISSUE THREAD

**Issue by StarsTesla** [oppo-us-research/NeuRBF#1]
Hi there, I notice that your model is running on A6000, how much GPU memory does it need? Cause I ran the code of NeRF follow the README, then OOM.

**Comment by LansburyCH** [comment]
Hi, StarsTesla, you can try reducing the batch_size_init parameter in the config files (configs/nerf_tensorf/nerf_synthetic_init.py and configs/nerf_tensorf/nerf_synthetic.py) to 2048 or smaller.

**Comment by StarsTesla** [comment]
Thanks about that, I successfully ran the code on my A100, which takes me over 35G memory, is there any idea about the huge usage of the memory on one single lego scene? I noticed that tensorf could run on V100(16G).

**Comment by LansburyCH** [comment]
This is a good question. We have not particularly optimized our codes for memory consumption. A possible reason for the high consumption is the feature interpolation process. This process comprises of: 1. for each point, get the features at neighboring RBFs; 2. for each point, compute the interpolation weights for each neighboring RBF; 3. multiply the features with interpolation weights, and then do sum reduction. Since our implementation is a PyTorch one (without customized CUDA kernel), this process will create some large intermediate tensors.

At the moment, changing the initial batch size (as mentioned in my previous reply) should be able to reduce memory consumption. This is because training takes more memory at the beginning than in the later stages (the masking mechanism in TensoRF does not come into effect until after 2000 steps).

**Comment by StarsTesla** [comment]
Thanks for your carefully reply!
